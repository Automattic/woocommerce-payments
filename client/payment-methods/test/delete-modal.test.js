/** @format */

/**
 * External dependencies
 */
import React from 'react';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';

/**
 * Internal dependencies
 */
import ConfirmPaymentMethodDeleteModal from '../delete-modal';
import paymentMethodsMap from '../../payment-methods-map';

const mockOnClose = jest.fn();
const mockOnConfirm = jest.fn();

describe( 'Activation Modal', () => {
	beforeAll( () => {
		wcSettings = {
			currentUserData: {
				email: 'foo@bar.com',
			},
		};
	} );
	afterEach( () => {
		jest.resetAllMocks();
	} );

	const renderDeleteModal = () => {
		return render(
			<ConfirmPaymentMethodDeleteModal
				id={ 'card' }
				label={ paymentMethodsMap.card.label }
				Icon={ paymentMethodsMap.card.Icon }
				onCancel={ mockOnClose }
				onConfirm={ mockOnConfirm }
			/>
		);
	};

	it( 'matches the snapshot', () => {
		const container = renderDeleteModal().baseElement;
		expect( container ).toMatchSnapshot();
	} );

	it( 'contains the label of the payment method', () => {
		renderDeleteModal();
		expect(
			screen.queryByText( paymentMethodsMap.card.label )
		).toBeInTheDocument();
	} );

	it( 'triggers the onClose event on close button click', () => {
		renderDeleteModal();
		const closeButton = screen.queryByRole( 'button', { name: 'Cancel' } );
		expect( mockOnClose ).not.toBeCalled();
		user.click( closeButton );
		expect( mockOnClose ).toBeCalled();
	} );

	it( 'triggers the onConfirmClose event on confirm button click', () => {
		renderDeleteModal();
		const confirmButton = screen.queryByRole( 'button', {
			name: 'Remove',
		} );
		expect( mockOnConfirm ).not.toBeCalled();
		user.click( confirmButton );
		expect( mockOnConfirm ).toBeCalled();
	} );
} );
