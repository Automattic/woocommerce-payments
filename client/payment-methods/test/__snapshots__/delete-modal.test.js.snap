// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activation Modal matches the snapshot 1`] = `
<body
  class="modal-open"
>
  <p
    aria-hidden="true"
    class="a11y-speak-intro-text"
    hidden="hidden"
    id="a11y-speak-intro-text"
    style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
  >
    Notifications
  </p>
  <div
    aria-atomic="true"
    aria-live="assertive"
    aria-relevant="additions text"
    class="a11y-speak-region"
    id="a11y-speak-assertive"
    style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
  />
  <div
    aria-atomic="true"
    aria-live="polite"
    aria-relevant="additions text"
    class="a11y-speak-region"
    id="a11y-speak-polite"
    style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
  />
  <div
    aria-hidden="true"
  />
  <div
    class="components-modal__screen-overlay"
  >
    <div
      aria-labelledby="components-modal-header-0"
      class="components-modal__frame wcpay-confirmation-modal"
      role="dialog"
      tabindex="-1"
    >
      <div
        class="components-modal__content"
        role="document"
      >
        <div
          class="components-modal__header"
        >
          <div
            class="components-modal__header-heading-container"
          >
            <h1
              class="components-modal__header-heading"
              id="components-modal-header-0"
            >
              Remove Credit / Debit card from checkout
            </h1>
          </div>
          <button
            aria-label="Close dialog"
            class="components-button has-icon"
            type="button"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 13.06l3.712 3.713 1.061-1.06L13.061 12l3.712-3.712-1.06-1.06L12 10.938 8.288 7.227l-1.061 1.06L10.939 12l-3.712 3.712 1.06 1.061L12 13.061z"
              />
            </svg>
          </button>
        </div>
        <div
          class="payment-delete-illustration__wrapper"
        >
          <div
            class="payment-delete-illustration__illustrations"
          >
            <img
              alt="Credit Card"
              class="payment-delete-illustration__payment-icon"
              src="assets/images/payment-methods/cc.svg"
            />
            <svg
              class="gridicon gridicons-cross-circle payment-delete-illustration__payment-cross-icon"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M19.1 4.9C15.2 1 8.8 1 4.9 4.9S1 15.2 4.9 19.1s10.2 3.9 14.1 0 4-10.3.1-14.2zm-4.3 11.3L12 13.4l-2.8 2.8-1.4-1.4 2.8-2.8-2.8-2.8 1.4-1.4 2.8 2.8 2.8-2.8 1.4 1.4-2.8 2.8 2.8 2.8-1.4 1.4z"
                />
              </g>
            </svg>
          </div>
        </div>
        <p>
          Are you sure you want to remove 
          <strong>
            Credit / Debit card
          </strong>
          ? Your customers will no longer be able to pay using Credit / Debit card.
        </p>
        <p>
          You can add it again at any time in 
          <a
            href="admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments"
          >
            WooPayments
          </a>
        </p>
        <hr
          class="wcpay-confirmation-modal__separator"
        />
        <div
          class="wcpay-confirmation-modal__footer"
        >
          <button
            class="components-button is-primary is-destructive"
            type="button"
          >
            Remove
          </button>
          <button
            class="components-button is-secondary"
            type="button"
          >
            Cancel
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
